datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -214,6 +214,9 @@ pub async fn from_substrait_plan(
                         Ok(from_substrait_rel(ctx, rel, &extensions).await?)
                     },
                     plan_rel::RelType::Root(root) => {
+                        if !root.names.is_empty() {
+                            extensions = extensions.with_projection_names(root.names.clone());
+                        }
```

Comment:
What's the point of storing the root names here? They only need to be applied to the final output, and that is handled below by the call to `make_renamed_schema`.